### PR TITLE
contrib/internal/httputil: return correct wrapped response writer

### PR DESCRIPTION
**Pull request by renanrt**
We **should** return the wrapped response writer and **NOT** the original one.

In response to [this](https://github.com/DataDog/dd-trace-go/issues/1076) issue.

### CODE REVIEW

File: contrib/internal/httputil/trace_gen.go [hunk 1]
```diff
@@ -8,9 +8,8 @@
 package httputil
 
 import (
-	"net/http"
-
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"net/http"
```
Comment (knusbaum) [inline contrib/internal/httputil/trace_gen.go:12]: This change should be reverted, but everything else looks alright.

**Review by knusbaum — CHANGES_REQUESTED** [review]
One nit about import order for the linter.

**Review by Julio-Guerra — CHANGES_REQUESTED** [review]
Thanks for contributing :-)

I just added the test that was missing and would have caught the issue you fix here.

The patch you propose is actually incorrect: `trace_gen.go` is a generated file and you should instead fix `make_responsewriter.go` and then run `go run ./make_responsewriter.go > trace_gen.go` to fix `trace_gen.go`.

Thank you :-)

File: contrib/internal/httputil/trace_gen.go [hunk 2]
```diff
@@ -125,6 +125,10 @@ func wrapResponseWriter(w http.ResponseWriter, span ddtrace.Span) http.ResponseW
 			monitoredResponseWriter
 			http.Hijacker
 		}{mw, hHijacker}
+	default:
+		w = struct {
+			monitoredResponseWriter
+		}{mw}
```
Comment (Julio-Guerra) [inline contrib/internal/httputil/trace_gen.go:131]: The following is enough:
```suggestion
		w = mw
```

File: contrib/internal/httputil/trace_gen.go [hunk 3]
```diff
@@ -125,6 +125,10 @@ func wrapResponseWriter(w http.ResponseWriter, span ddtrace.Span) http.ResponseW
 			monitoredResponseWriter
 			http.Hijacker
 		}{mw, hHijacker}
+	default:
```
Comment (gbbr) [inline contrib/internal/httputil/trace_gen.go:128]: This is generated code. If you want changes, you have to modify the code that generates this file. 

https://github.com/DataDog/dd-trace-go/blob/a71ee32b5e69efa07207a5959df796f157ba410c/contrib/internal/httputil/trace_gen.go#L6

**Review by gbbr — CHANGES_REQUESTED** [review]

**Comment by renanrt** [comment]
Updated the file that generates the code, tested and it seems to be working fine!

**Review by Julio-Guerra — APPROVED** [review]

**Pull request merged**